src/gamma_inc.jl

```diff
@@ -0,0 +1,577 @@
+        throw(DomainError((a,x,ind,"P(a,x) or Q(a,x) is computationally indeterminant in this case.")))
+    end
+    c = 1.0 - y
+    w = (0.5 - sqrt(y)*(0.5 + (0.5 - y/3.0))/rtpi)/c
+    u = 1.0/a
+    if iop == 1
+        c0 = @horner(z , d00 , d0[1] , d0[2] , d0[3])
+        c1 = @horner(z , d10 , d1[1] , d1[2] , d1[3])
+        c2 = @horner(z , d20 , d2[1] , d2[2])
+        c3 = @horner(z , d30 , d3[1] , d3[2])
+        c4 = @horner(z , d40 , d4[1])
+        c5 = @horner(z , d50 , d5[1])
+        c6 = @horner(z , d60 , d6[1])
+
+        t = @horner(u , c0 , c1 , c2 , c3 , c4 , c5 , c6 , d70 , d80)
+        
+    elseif iop == 2
+        c0 = @horner(d00 , d0[1] , d0[2])
+        c1 = @horner(d10 , d1[1])
+        t = @horner(u , c0 , c1 , d20)
+    
+    else
+        t = @horner(z , d00 , d0[1])
+        
+    end
+    if l < 1.0
+        return c*(w - rt2pin*t/sqrt(a))
+    end
+    return 1.0 - c*(w + rt2pin*t/sqrt(a))
+end
+"""
+    gamma_p_fsum(a,x)
+
+Compute using Finite Sums for Q(a,x) when a >= 1 && 2a is integer
+"""
+function gamma_p_fsum(a::Float64, x::Float64)
+    if isinteger(a)           
+        sm = exp(-x)
+        t = sm
+        N = 1
+        c=0.0
+        i = trunc(Int,a - 0.5)
+    else
+        rtx = sqrt(x)
+        sm = erfc(rtx)
+        t = exp(-x)/(rtpi*rtx)
+        N=0
+        c=-0.5
+        i = trunc(Int,a - 0.5)
+    end
+    for lp = N : i
+        if i == 0
+            break
+        end
+        c += 1.0
+        t = (x*t)/c
+        sm += t
+    end
+    return 1.0 - sm
+
+end
+# Reference : 'Computation of the incomplete gamma function ratios and their inverse' by Armido R DiDonato , Alfred H Morris.
+# Published in Journal: ACM Transactions on Mathematical Software (TOMS)
+# Volume 12 Issue 4, Dec. 1986 Pages 377-393
+# doi>10.1145/22721.23109
+
+"""
+    gamma_p(a,x,IND)
+    
+    DLMF: https://dlmf.nist.gov/8.2#E4 , https://dlmf.nist.gov/8.2#E5
+    Wiki: https://en.wikipedia.org/wiki/Incomplete_gamma_function
+    IND --> Accuracy desired ; IND=0 means 14 significant digits accuracy , IND=1 means 6 significant digit and IND=2 means only 3 digit accuracy suffices.
+    gamma_p(a,x) or P(a,x) is the Incomplete gamma function ratio given by : ``1/\\Gamma (a) \\int_{0}^{x} e^{-t}t^{a-1} dt``
+"""
+function gamma_p(a::Float64,x::Float64,ind::Integer)
+    iop = ind + 1
+    acc = acc0[iop]
+    if a<0.0 || x<0.0
+        throw(DomainError((a,x,ind,"`a` and `x` must be greater than 0 ---- Domain : (0,inf)")))
+    elseif a==0.0 && x==0.0
+        throw(DomainError((a,x,ind,"`a` and `x` must be greater than 0 ---- Domain : (0,inf)")))
+    elseif a*x==0.0 
+        if x<=a
+            return 0.0
+        else
+            return 1.0
+        end               
+    end
+    
+    if a >= 1.0
+        if a >= big1[iop]
+            @goto l20
+        elseif a > x || x >= x0[iop] || isinteger(2*a)  
+            @goto l30 
+       #else
+       #    return gamma_p_fsum(a,x)
```

Comment:
what happened here?

Reply:
This is possibly an error in the Didonato paper. The commented section is according to the approximation algorithm given in the paper. But it breaks tests.
I have changed that to what it is now, and it's passing tests too and the relative error plot as you saw was fine too.